### PR TITLE
[csharp-nancyfx] Delete NancyFX

### PR DESCRIPTION
**Pull request by devhl-labs**
NancyFX is archived and the last commit was two years ago.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02) @Blackclaws (2021/03) @lucamazzanti (2021/05)

### CODE REVIEW

**Comment by wing328** [comment]
@devhl-labs thanks for the PR. I've filed https://github.com/OpenAPITools/openapi-generator/pull/11454 to start with and we will remove it in the 6.x release.

**Comment by devhl-labs** [comment]
Can you give this one a milestone so it is not missed?

**Comment by wing328** [comment]
Can you please resolve the merge conflicts when you've time? Upcoming version is 6.0.0 (master) so we can remove this generator.

File: README.md [hunk 1]
```diff
@@ -949,7 +949,6 @@ Here is a list of template creators:
    * C# ASP.NET Core 3.0: @A-Joshi
    * C# APS.NET Core 3.1: @phatcher
    * C# Azure functions: @Abrhm7786
-   * C# NancyFX: @mstefaniuk
```
Comment (wing328) [inline README.md:952]: I'll add it back after merging it into master as old versions still have this generator available.

**Pull request merged**